evm/generators.go

```diff
@@ -1281,131 +1598,193 @@ func {{.HandlerName}}() *cobra.Command {
+                }
+                if safeApi == "" {
+					client, clientErr := NewClient(rpc)
+					if clientErr != nil {
+						return clientErr
+					}
+					chainIDCtx, cancelChainIDCtx := NewChainContext(timeout)
+					defer cancelChainIDCtx()
+					chainID, chainIDErr := client.ChainID(chainIDCtx)
+					if chainIDErr != nil {
+						return chainIDErr
+					}
+					safeApi = "https://safe-client.safe.global/v1/chains/" + chainID.String() + "/transactions/" + safeAddress + "/propose"
+					fmt.Println("--safe-api not specified, using default (", safeApi, ")")
 				}
 
-				gasEstimationCtx, cancelGasEstimationCtx := NewChainContext(timeout)
-				defer cancelGasEstimationCtx()
-
-				gasEstimate, gasEstimateErr := client.EstimateGas(gasEstimationCtx, estimationMessage)
-				if gasEstimateErr != nil {
-					return gasEstimateErr
+				if OperationType(safeOperation).String() == "Unknown" {
+					return fmt.Errorf("--safe-operation must be 0 (Call) or 1 (DelegateCall)")
 				}
-
-				transactionBinary, transactionBinaryErr := transaction.MarshalBinary()
-				if transactionBinaryErr != nil {
-					return transactionBinaryErr
+            }
+
+            {{range .MethodArgs}}
+            {{.PreRunE}}
+            {{- end}}
+
+            return nil
+        },
+        RunE: func(cmd *cobra.Command, args []string) error {
+            client, clientErr := NewClient(rpc)
+            if clientErr != nil {
+                return clientErr
+            }
+
+            key, keyErr := KeyFromFile(keyfile, password)
+            if keyErr != nil {
+                return keyErr
+            }
+
+            chainIDCtx, cancelChainIDCtx := NewChainContext(timeout)
+            defer cancelChainIDCtx()
+            chainID, chainIDErr := client.ChainID(chainIDCtx)
+            if chainIDErr != nil {
+                return chainIDErr
+            }
+
+            transactionOpts, transactionOptsErr := bind.NewKeyedTransactorWithChainID(key.PrivateKey, chainID)
+            if transactionOptsErr != nil {
+                return transactionOptsErr
+            }
+
+            SetTransactionParametersFromArgs(transactionOpts, nonce, value, gasPrice, maxFeePerGas, maxPriorityFeePerGas, gasLimit, simulate)
+
+            contract, contractErr := New{{$structName}}(contractAddress, client)
+            if contractErr != nil {
+                return contractErr
+            }
+
+            session := {{$structName}}TransactorSession{
+                Contract: &contract.{{$structName}}Transactor,
+                TransactOpts: *transactionOpts,
+            }
+
+            if safeAddress != "" {
+                // Generate transaction data
+				abi, err := {{$structName}}MetaData.GetAbi()
+                if err != nil {
+                    return fmt.Errorf("failed to get ABI: %v", err)
+                }
+				// TODO: this is a workaround to match the original method name in the ABI, need to fix to work with every method name
```

Comment:
What's the issue here?

Reply:
I will try my best to explain here. If it's not clear, maybe we should pair for a better understanding:

I found out that `abi.Pack("methodName", args...)` is case sensitive. It only works if the `methodName` has the exact case-sensitive name as in the abi. Unfortunately,  the parser modifies the method name to `UpperCamelCase`.

I've tried a similar method of calling the `transactor` with `NoSend` flag equal to `true`. However, it does not work because it simulates the transaction against the blockchain. In this case, the dev wallet is the `from` address, and it needs to be the `safeAddress`. I couldn't find an option to the the `from` address as a different address from the caller. 

I think there is probably a more straightforward solution, so I left this `TODO:` comment so we could work together on this.

Reply:
Could we put the original method name into the object that gets sent to the template?

Reply:
Just fixed the problem using `session.` object. It create the calldata without sending it